src/moscot/plotting/_utils.py

```diff
@@ -359,45 +360,137 @@ def _plot_temporal(
     adata: AnnData,
     temporal_key: str,
     key_stored: str,
-    time_points: Optional[Iterable[K]] = None,
+    start: float,
+    end: float,
+    categories: Optional[Union[str, List[str]]] = None,
+    *,
+    push: bool,
+    time_points: Optional[Sequence[K]] = None,
     basis: str = "umap",
-    result_key: str = "plot_tmp",
-    constant_fill_value: float = 0.0,
+    constant_fill_value: float = np.nan,
+    scale: bool = True,
     cont_cmap: Union[str, mcolors.Colormap] = "viridis",
-    title: Optional[str] = None,
+    title: Optional[Union[str, List[str]]] = None,
+    suptitle: Optional[str] = None,
     figsize: Optional[Tuple[float, float]] = None,
     dpi: Optional[int] = None,
+    dot_scale_factor: float = 2.0,
+    na_color: Optional[str] = "#e8ebe9",
     save: Optional[str] = None,
     ax: Optional[Axes] = None,
     show: bool = False,
+    suptitle_fontsize: Optional[float] = None,
     **kwargs: Any,
-) -> mp.figure.Figure:
-    all_keys = adata.obs[temporal_key].unique()
-    if time_points is None:
-        constant_fill_keys: Set[K] = set()
+) -> mpl.figure.Figure:
+
+    fig, axs = plt.subplots(
+        1, 1 if time_points is None else len(time_points), figsize=figsize, dpi=dpi, constrained_layout=True
+    )
+    axs = np.ravel(axs)  # make into iterable
+
+    if not push and time_points is not None:
+        time_points = time_points[::-1]
+
+    if isinstance(title, list):
+        if TYPE_CHECKING:
+            assert isinstance(time_points, list)
+        if len(title) != len(time_points):
+            raise ValueError("If `title` is a list, its length must be equal to the length of `time_points`.")
+        titles = title
     else:
-        constant_fill_keys = set(all_keys) - set(time_points)
-    tmp = np.full(len(adata), np.nan)
-    for t in adata.obs[temporal_key].unique():
-        mask = adata.obs[temporal_key] == t
-        if t in constant_fill_keys:
-            tmp[mask] = constant_fill_value
+        name = "descendants" if push else "ancestors"
+        if time_points is not None:
+            titles = [f"{categories} at time point {start if push else end}"]
```

Comment:
Feel like the titles might be too long, but not sure. Maybe remove `time point`.